tpl/tplimpl/template.go

```diff
@@ -767,6 +767,7 @@ func (t *templateHandler) postTransform() error {
 	defineCheckedHTML := false
 	defineCheckedText := false
 
+	t.main.mu.RLock()
```

Comment:
There's a return in the following for-loop where this lock is not released.  Can you use `defer` here?

Reply:
You are right! I've overlooked that one. I don't see that operations below are time-consuming so I agree that defer is preferable over having two unlocks.

Reply:
No, actually I think we can't use it because of `t.extractPartials(v.Template)` using the lock inside. I assume tests will highlight that and I'll need to change it.

Reply:
@moorereason what do you think now?